plugins/router/freebox

```diff
@@ -0,0 +1,316 @@
+			echo 'noise_down.graph no'
+			echo 'noise.label Noise'
+			echo 'noise.negative noise_down'
+
+			echo 'attenuation_down.label Attenuation down'
+			echo 'attenuation_down.graph no'
+			echo 'attenuation.label Attenuation'
+			echo 'attenuation.negative attenuation_down'
+			;;
+		.adsl_errors)
+			echo 'graph_title ADSL error correction'
+			echo 'graph_category network'
+			echo 'graph_vlabel errors in (+) / out (-)'
+			echo 'graph_order fec_down fec crc_down crc hec_down hec'
+
+			echo 'fec_down.label FEC down'
+			echo 'fec_down.graph no'
+			echo 'fec.label FEC'
+			echo 'fec.negative fec_down'
+
+			echo 'crc_down.label CRC down'
+			echo 'crc_down.graph no'
+			echo 'crc.label CRC'
+			echo 'crc.negative crc_down'
+
+			echo 'hec_down.label HEC down'
+			echo 'hec_down.graph no'
+			echo 'hec.label HEC'
+			echo 'hec.negative hec_down'
+			;;
+		.traffic)
+			echo 'graph_title Traffic'
+			echo 'graph_category network'
+			echo 'graph_vlabel bits per second in (+) / out (-)'
+			echo 'graph_order atm_down atm wan_down wan eth_down eth usb_down usb switch_down switch'
+
+			echo 'atm_down.label ATM down'
+			echo 'atm_down.graph no'
+			echo 'atm_down.cdef atm_down,1000,*'
+			echo 'atm.label ATM sync'
+			echo 'atm.negative atm_down'
+			echo 'atm.cdef atm,1000,*'
+
+			echo 'wan_down.label WAN down'
+			echo 'wan_down.graph no'
+			echo 'wan_down.cdef wan_down,8000,*'
+			echo 'wan.label WAN'
+			echo 'wan.negative wan_down'
+			echo 'wan.cdef wan,8000,*'
+
+			echo 'eth_down.label ETH down'
+			echo 'eth_down.graph no'
+			echo 'eth_down.cdef eth_down,8000,*'
+			echo 'eth.label Ethernet'
+			echo 'eth.negative eth_down'
+			echo 'eth.cdef eth,8000,*'
+
+			echo 'usb_down.label USB down'
+			echo 'usb.label USB'
+			echo 'usb_down.graph no'
+			echo 'usb_down.cdef usb_down,8000,*'
+			echo 'usb.negative usb_down'
+			echo 'usb.cdef usb,8000,*'
+
+			echo 'switch_down.label Switch down'
+			echo 'switch_down.graph no'
+			echo 'switch_down.cdef switch_down,8000,*'
+			echo 'switch.label Switch'
+			echo 'switch.cdef switch,8000,*'
+			echo 'switch.negative switch_down'
+			;;
+		.uptime)
+			echo 'graph_title Uptimes'
+			echo 'graph_category network'
+			echo 'graph_vlabel days'
+			echo 'graph_args --logarithmic'
+
+			echo 'freebox.label Freebox'
+			echo 'freebox.draw AREA'
+			echo 'freebox.cdef freebox,86400,/'
+
+			echo 'connection.label Connection'
+			echo 'connection.cdef connection,86400,/'
+			;;
+		.users)
+			echo 'graph_title Network users'
+			echo 'graph_category network'
+			echo 'graph_vlabel count'
+
+			echo 'dhcp.label DHCP leases'
+			echo 'redirect_tcp.label TCP redirections'
+			echo 'redirect_udp.label UDP redirections'
+			;;
+		*)
+			echo 'graph_title Freebox connection'
+			echo 'graph_category network'
+			echo 'graph_vlabel bits per second in (+) / out (-)'
+			echo 'graph_order main_atm_down main_atm main_wan_down main_wan'
+
+			# XXX: parameters duplicated from traffic
```

Comment:
Why are these duplicated?

Reply:
It's a multigraph, so I have those both in the traffic subgraph, and the main graph.

I'm curious to know if there is any other way to do this.

Reply:
Ah - ok. I did not get, that the duplication is in the main graph. This is indeed fine.